cadc-vos/src/main/java/ca/nrc/cadc/vos/client/VOSpaceClient.java

```diff
@@ -629,8 +634,15 @@ private ClientTransfer createTransferSync(Transfer transfer)
             	log.debug("Quick transfer URL: " + redirectURL);
             	// create a new transfer with a protocol with an end point
             	List<Protocol> prots = new ArrayList<Protocol>();
-            	prots.add(new Protocol(transfer.getProtocols().iterator().next().getUri(), redirectURL.toString(), null));
-            	Transfer trf = new Transfer(transfer.getTarget(), transfer.getDirection(), prots);
+            	prots.add(new Protocol(transfer.protocols.iterator().next().getUri(), redirectURL.toString(), null));
+
+            	// Assumption: quick transfer will be a single target
+                if (transfer.targets == null || transfer.targets.size() != 1) {
+                    throw new IllegalArgumentException("Quick transfer only supports single targets.");
+                }
+
```

Comment:
This isn't a 'quick transfer' here so this code shouldn't be here?

Reply:
Both of these 'Quick transfer' IllegalArgumentException throws are in stanzas starting with: 
            if( transfer.isQuickTransfer()) {